main.cpp

```diff
@@ -1,7 +1,335 @@
 #include <iostream>
+#include <fstream>
+#include <vector>
+#include <stdlib.h>
+#include <time.h>
+#include <math.h>
+#include <bits/stdc++.h> 
+#include <random> 
+#include <chrono>   
+#include <algorithm>
+
+
 using namespace std;
-int main()
-{
-    cout << "Hello, World!";
+
+
+class MachineBlock{    
+    int start;
+    int end;
+    int taskNumber;
+    int operationNumber;
+    string blockType;
+    public:
+    void set_start(int st){
+        start = st;
+    }
+    void set_end(int en){
+        end = en;
+    }
+    void set_taskNumber(int tN){
+        taskNumber = tN;
+    }
+    void set_operationNumber(int oN){
+        operationNumber = oN;
+    }
+    void set_blockType(string bT){
+        blockType = bT;
+    }
+    int get_start(){
+        return start;
+    }
+    int get_end(){
+        return end;
+    }
+    int get_taskNumber(){
+        return taskNumber;
+    }
+    int get_operationNumber(){
+        return operationNumber;
+    }
+    string get_blockType(){
+        return blockType;
+    }
+};
+
+
+class Task{
+    int taskNumber;
+    int operation1;
+    int operation2;
+    public:
+    void set_task_number(int tN){
+        taskNumber = tN;
+    }
+    void set_op1(int op1){  
+        operation1 = op1;
+    }
+    void set_op2(int op2){
+        operation2 = op2;
+    }
+    int get_task_number(){
+        return taskNumber;
+    }
+    int get_op1(){
+        return operation1;
+    }
+    int get_op2(){
+        return operation2;
+    }
+};
+
+
+void addBlock(vector <MachineBlock> &machine, const int start, const int end, const string type, const int taskNumber = 0, const int operationNumber = 0, const int maintenance = 0, const int taskLength = 0, const int maxTimeWork = 0, const float latency = 0);
+
+
+void readFile(const string fileName, vector <Task> &tasks){
+    string line;
+    ifstream input (fileName);
+    if (input.is_open()){
+        while (getline (input, line)){
+            int taskNumber = stoi(line.substr(line.find('t') +1,line.find(':') -1));
+            string operations = line.substr(line.find(' ') +1,line.find(','));
+            int operation1 = stoi(operations.substr(0,operations.find(',')));
+            int operation2 = stoi(operations.substr(operations.find(',') +1,operations.size()));
+            Task obj;
+            obj.set_task_number(taskNumber);
+            obj.set_op1(operation1);
+            obj.set_op2(operation2);
+            tasks.push_back(obj);
+        }
+        input.close();
+    }
+}
+
+ 
+int getCmax(vector <MachineBlock> &machine){
+    if (machine.size() != 0)
+            return machine[machine.size() -1].get_end();
```

Comment:
in this method you are calculating machine.size() twice. It would be better to create variable at the beginning of method.

> machineCount = machine.size();